include/net/nrf_cloud_location.h

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef NRF_CLOUD_LOCATION_H_
+#define NRF_CLOUD_LOCATION_H_
+
+/** @file nrf_cloud_location.h
+ * @brief Module to provide nRF Cloud location support to nRF9160 SiP.
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/net/wifi.h>
+#include <modem/lte_lc.h>
+#include <net/nrf_cloud.h>
+#include <cJSON.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @defgroup nrf_cloud_location nRF Cloud location
+ * @{
+ */
+
+/* 2 chars per byte, colon separated */
+#define WIFI_MAC_ADDR_STR_LEN ((WIFI_MAC_ADDR_LEN * 2) + 5)
+
+/** @defgroup nrf_cloud_location_wifi_omit Omit item from Wi-Fi location request.
+ * @{
+ */
+#define NRF_CLOUD_LOCATION_WIFI_OMIT_RSSI	(INT8_MAX)
+#define NRF_CLOUD_LOCATION_WIFI_OMIT_CHAN	(0)
+/** @} */
+
+struct wifi_ap_info {
+	char mac_addr_str[WIFI_MAC_ADDR_STR_LEN + 1];
+	char ssid_str[WIFI_SSID_MAX_LEN + 1];
+	uint8_t channel;
+	int8_t rssi;
+};
+
+/* Minimum number of access points required by nRF Cloud */
+#define NRF_CLOUD_LOCATION_WIFI_AP_CNT_MIN 2
+struct wifi_scan_result {
+	struct wifi_ap_info *ap_info;
+	uint8_t cnt;
+};
+
+/** @brief Location request type */
+enum nrf_cloud_location_type {
+	LOCATION_TYPE_SINGLE_CELL,
+	LOCATION_TYPE_MULTI_CELL,
+	LOCATION_TYPE_WIFI,
+
+	LOCATION_TYPE__INVALID
+};
+
+/** @brief Location request result */
+struct nrf_cloud_location_result {
+	enum nrf_cloud_location_type type;
+	double lat;
+	double lon;
+	uint32_t unc;
+	/* Error value received from nRF Cloud */
+	enum nrf_cloud_error err;
```

Comment:
It would be nice to have some descriptions for these, at least for `unc`. Might also be good to mention the value of `err` in case the result was success (probably `NRF_CLOUD_ERR_STATUS_NONE`?).